docker/docker_transport_test.go

```diff
@@ -43,17 +44,22 @@ func TestParseReference(t *testing.T) {
 
 // testParseReference is a test shared for Transport.ParseReference and ParseReference.
 func testParseReference(t *testing.T, fn func(string) (types.ImageReference, error)) {
-	for _, c := range []struct{ input, expected string }{
-		{"busybox", ""}, // Missing // prefix
-		{"//busybox:notlatest", "docker.io/library/busybox:notlatest"},           // Explicit tag
-		{"//busybox" + sha256digest, "docker.io/library/busybox" + sha256digest}, // Explicit digest
-		{"//busybox", "docker.io/library/busybox:latest"},                        // Default tag
+	for _, c := range []struct {
+		input, expected       string
+		expectedUnknownDigest bool
+	}{
+		{"busybox", "", false}, // Missing // prefix
+		{"//busybox:notlatest", "docker.io/library/busybox:notlatest", false},           // Explicit tag
+		{"//busybox" + sha256digest, "docker.io/library/busybox" + sha256digest, false}, // Explicit digest
+		{"//busybox", "docker.io/library/busybox:latest", false},                        // Default tag
 		// A github.com/distribution/reference value can have a tag and a digest at the same time!
 		// The docker/distribution API does not really support that (we can’t ask for an image with a specific
 		// tag and digest), so fail.  This MAY be accepted in the future.
-		{"//busybox:latest" + sha256digest, ""},                                    // Both tag and digest
-		{"//docker.io/library/busybox:latest", "docker.io/library/busybox:latest"}, // All implied values explicitly specified
-		{"//UPPERCASEISINVALID", ""},                                               // Invalid input
+		{"//busybox:latest" + sha256digest, "", false},                                     // Both tag and digest
+		{"//docker.io/library/busybox:latest", "docker.io/library/busybox:latest", false},  // All implied values explicitly specified
+		{"//UPPERCASEISINVALID", "", false},                                                // Invalid input
+		{"//busybox" + UnknownDigestSuffix, "docker.io/library/busybox", true},             // UnknownDigest suffix
```

Comment:
(Non-blocking: throughout the tests, hard-coding the `@@…@@` string, instead of referring to the same variable the primary code uses, would ensure the tests detect if we ever accidentally changed the variable value.)

Reply:
Ack, fixed